wagtail/core/models.py

```diff
@@ -1830,8 +1976,45 @@ def is_latest_revision(self):
         latest_revision = PageRevision.objects.filter(page_id=self.page_id).order_by('-created_at', '-id').first()
         return (latest_revision == self)
 
-    def publish(self):
+    def publish(self, user=None, changed=True, log_action=True, previous_revision=None):
+        """
+        Publishes or schedules revision for publishing.
+
+        :param user: the publishing user
+        :param changed: indicated whether content has changed
+        :param log_action:
+            flag for the logging action. Pass False to skip logging. Cannot pass an action string as the method
+            performs several actions: "publish", "revert" (and publish the reverted revision),
+            "schedule publishing with a live revision", "schedule revision reversal publishing, with a live revision",
+            "schedule publishing", "schedule revision reversal publishing"
+        :param previous_revision: indicates a revision reversal. Should be set to the previous revision instance
+        """
         page = self.as_page_object()
+
+        def log_scheduling_action(revision, previous_revision=None, user=None, changed=changed):
+            if not previous_revision:
+                action = 'wagtail.schedule.publish'
```

Comment:
When I first saw this, I assumed it meant "published using a wagtail schedule". Could we rename this to `wagtail.schedule`?

Reply:
renamed to `wagtail.publish.schedule` as per Slack convo